cardinal/ecs/search.go

```diff
@@ -34,27 +35,42 @@ func NewSearch(filter filter.ComponentFilter) *Search {
 	}
 }
 
+func (q *Search) Commit() error {
+	var err error
+	for _, callback := range q.callbacks {
+		err = callback()
+		if err != nil {
+			break
+		}
+	}
+	q.callbacks = q.callbacks[:0]
```

Comment:
There's alternative behavior here to make the value "consumed". Once committed it will always return an error. lmk if you prefer that, I don't really care.